src/Storages/NamedCollections/NamedCollectionUtils.cpp

```diff
@@ -78,7 +78,7 @@ class LoadFromConfig
         /// (`enumerate_result` == <collection_path>).
         const bool collection_is_empty = enumerate_result.size() == 1
             && *enumerate_result.begin() == collection_prefix;
-        std::set<std::string> keys;
+        std::set<std::string, std::less<>> keys;
```

Comment:
Why the order is important here?

Reply:
I'll fix in the next PR. Do not remember for sure, but `unordered_set` should be ok too.